include/geos/operation/distance/IndexedFacetDistance.h

```diff
@@ -24,19 +24,80 @@
 namespace geos {
 namespace operation {
 namespace distance {
+
+/// \brief Computes the distance between the facets (segments and vertices)
+/// of two [Geometrys](\ref geom::Geometry) using a Branch-and-Bound algorithm.
+///
+/// The Branch-and-Bound algorithm operates over a traversal of R-trees built
+/// on the target and the query geometries.
+///
+/// This approach provides the following benefits:
+///
+/// - Performance is dramatically improved due to the use of the R-tree index
+///   and the pruning due to the Branch-and-Bound approach
+/// - The spatial index on the target geometry is cached which allow reuse in
+///   an repeated query situation.
+///
+/// Using this technique is usually much more performant than using the
+/// brute-force \ref geom::Geometry::distance(const Geometry* g) const when one
+/// or both input geometries are large, or when evaluating many distance
+/// computations against a single geometry.
+///
+/// This class is thread-safe.
```

Comment:
As copied from JTS. But is this the case for GEOS too?

Reply:
Maybe? But I don't think we need to make that claim.

Reply:
Ok, I removed that line.